File: src/core/ai-session.js
import { attachTeamRoles, buildAccountabilityStrip } from './team-roles.js';

const SESSION_VERSION = 'jericho-ai-session-v1';

export function buildSessionSnapshot({
  state,
  pipelineOutput,
  scene,
  narrative,
  directives,
  commandSpec,
  reasoning,
  chain,
  multiGoal,
  integrityDeviations,
  teamRoles,
  accountabilityStrip
}) {
  const team = state?.team || {};
  const teamIdentity = pipelineOutput?.analysis?.teamIdentity || null;
  const teamGovernance = pipelineOutput?.analysis?.teamGovernance || null;
  const teamNarrative = pipelineOutput?.analysis?.teamNarrative || null;
  const resolvedTeamRoles = teamRoles || attachTeamRoles(team, pipelineOutput);

  const snapshot = {
    version: SESSION_VERSION,
    state,
    teamIdentity: teamIdentity,
    teamGovernance: teamGovernance,
    teamNarrative: teamNarrative,
    teamRoles: resolvedTeamRoles,
    team: {
      users: team.users || [],
      teams: team.teams || [],
      roles: team.roles || {},
      goalsById: team.goalsById || {},
      teamCycles: team.teamCycles || {}
    },
    analysis: {
      pipeline: pipelineOutput,
      narrative,
      directives: {
        list: directives?.directives || [],
        summary: directives?.summary || ''
      },
      scene,
      reasoning: reasoning || null,
      chain: chain || null,
      multiGoal: multiGoal || null,
      integrityDeviations: integrityDeviations || null,
      teamIdentity,
      teamGovernance,
      teamNarrative
    },
    meta: {
      commands: commandSpec || {},
      invariants: {
        deterministic: true,
        readOnly: true
      }
    }
  };

  snapshot.accountabilityStrip =
    accountabilityStrip || buildAccountabilityStrip(snapshot, resolvedTeamRoles);

  return snapshot;
}

export default { buildSessionSnapshot };

File: backend/jyriko/domain/ai_session.py
"""Stub for removed ai_session — v1 team system."""

import uuid
from typing import Any


def build_session_snapshot(
    state: dict[str, Any],
    pipeline_output: dict[str, Any],
    scene: dict[str, Any],
    narrative: dict[str, Any],
    directives: dict[str, Any],
    reasoning: dict[str, Any],
    chain: dict[str, Any],
    multi_goal: dict[str, Any],
    integrity_deviations: dict[str, Any],
) -> dict[str, Any]:
    """Build session snapshot."""
    return {
        "id": str(uuid.uuid4()),
        "goal": (pipeline_output or {}).get("goal"),
        "integrity": ((pipeline_output or {}).get("integrity") or {}).get("score", 0),
        "tasks": (pipeline_output or {}).get("tasks") or [],
        "narrative": narrative.get("text"),
        "directives": directives.get("directives"),
        "teamRoles": {},
    }
